{"schemaVersion":{"major":0,"patch":0,"minor":3},"metadata":{"symbolKind":"method","roleHeading":"Instance Method","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"text":"awaitWithoutException","kind":"identifier"},{"kind":"text","text":"("},{"text":"millis","kind":"externalParam"},{"kind":"text","text":": "},{"text":"Int32","preciseIdentifier":"s:s5Int32V","kind":"typeIdentifier"},{"text":") -> ","kind":"text"},{"text":"Bool","kind":"typeIdentifier","preciseIdentifier":"s:Sb"}],"title":"awaitWithoutException(millis:)","externalID":"s:15DXFeedFramework7PromiseC21awaitWithoutException6millisSbs5Int32V_tF","role":"symbol","modules":[{"name":"DXFeedFramework"}]},"abstract":[{"text":"Wait for computation to complete and return its result or throw an exception in case of exceptional completion.","type":"text"},{"type":"text","text":" "},{"type":"text","text":"If the wait times out, then the computation is "},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()"},{"text":" cancelled and exception is thrown.","type":"text"}],"identifier":{"url":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/awaitWithoutException(millis:)","interfaceLanguage":"swift"},"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/dxfeedframework\/promise\/awaitwithoutexception(millis:)"]}],"sections":[],"hierarchy":{"paths":[["doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework","doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise"]]},"kind":"symbol","primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["macOS"],"tokens":[{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"awaitWithoutException","kind":"identifier"},{"kind":"text","text":"("},{"text":"millis","kind":"externalParam"},{"kind":"text","text":" "},{"kind":"internalParam","text":"timeOut"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","text":"Int32","preciseIdentifier":"s:s5Int32V"},{"text":") -> ","kind":"text"},{"preciseIdentifier":"s:Sb","text":"Bool","kind":"typeIdentifier"}],"languages":["swift"]}]},{"kind":"content","content":[{"type":"heading","level":2,"text":"Return Value","anchor":"return-value"},{"inlineContent":[{"text":"If the wait times out, then the computation is ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()","type":"reference","isActive":true},{"text":" cancelled and this method returns ","type":"text"},{"inlineContent":[{"type":"text","text":"false"}],"type":"strong"},{"type":"text","text":"."},{"text":" ","type":"text"},{"type":"text","text":"Use this method in the code that shall continue normal execution in case of timeout."}],"type":"paragraph"}]}],"references":{"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise":{"title":"Promise","type":"topic","fragments":[{"text":"class","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"Promise"}],"role":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise","url":"\/documentation\/dxfeedframework\/promise","navigatorTitle":[{"kind":"identifier","text":"Promise"}],"abstract":[{"text":"Result of a computation that will be completed normally or exceptionally in the future.","type":"text"},{"text":" ","type":"text"},{"text":"This class is designed to represent a promise to deliver certain result.","type":"text"}],"kind":"symbol"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework":{"abstract":[],"url":"\/documentation\/dxfeedframework","kind":"symbol","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework","type":"topic","title":"DXFeedFramework","role":"collection"},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/isDone()":{"url":"\/documentation\/dxfeedframework\/promise\/isdone()","role":"symbol","title":"isDone()","type":"topic","kind":"symbol","abstract":[{"type":"text","text":"Returns "},{"type":"strong","inlineContent":[{"type":"text","text":"true"}]},{"type":"text","text":" when computation has"},{"text":" ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/complete(result:)","isActive":true,"type":"reference"},{"text":" completed normally,","type":"text"},{"text":" ","type":"text"},{"text":"or ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/completeExceptionally(_:)","isActive":true,"type":"reference"},{"text":" exceptionally,","type":"text"},{"text":" ","type":"text"},{"type":"text","text":"or was "},{"isActive":true,"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()","type":"reference"},{"text":" canceled","type":"text"}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"kind":"identifier","text":"isDone"},{"text":"() -> ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:Sb","text":"Bool"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/cancel()":{"url":"\/documentation\/dxfeedframework\/promise\/cancel()","role":"symbol","title":"cancel()","type":"topic","kind":"symbol","abstract":[{"text":"Cancels computation. This method does nothing if computation has already ","type":"text"},{"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/isDone()","isActive":true,"type":"reference"},{"text":" completed.","type":"text"}],"identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()","fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"()"}]},"doc://com.dxfeed.DXFeedFramework/documentation/DXFeedFramework/Promise/awaitWithoutException(millis:)":{"fragments":[{"kind":"keyword","text":"func"},{"text":" ","kind":"text"},{"text":"awaitWithoutException","kind":"identifier"},{"text":"(","kind":"text"},{"kind":"externalParam","text":"millis"},{"text":": ","kind":"text"},{"text":"Int32","preciseIdentifier":"s:s5Int32V","kind":"typeIdentifier"},{"kind":"text","text":") -> "},{"text":"Bool","kind":"typeIdentifier","preciseIdentifier":"s:Sb"}],"abstract":[{"text":"Wait for computation to complete and return its result or throw an exception in case of exceptional completion.","type":"text"},{"type":"text","text":" "},{"type":"text","text":"If the wait times out, then the computation is "},{"isActive":true,"type":"reference","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/cancel()"},{"type":"text","text":" cancelled and exception is thrown."}],"url":"\/documentation\/dxfeedframework\/promise\/awaitwithoutexception(millis:)","role":"symbol","kind":"symbol","type":"topic","identifier":"doc:\/\/com.dxfeed.DXFeedFramework\/documentation\/DXFeedFramework\/Promise\/awaitWithoutException(millis:)","title":"awaitWithoutException(millis:)"}}}